api/v1/utils/response.js

```diff
@@ -0,0 +1,23 @@
+const commonUtils = require("./common");
+
+class SuccessResponse {
```

Comment:
t nghĩ cái successResponse này ổn k, ổn thì t làm 1 cái Error Response, rồi ném cái ErrorLogic vào đây luôn. @kc97ble

Reply:
Không nên làm như vậy. Đây là [over-engineering](https://en.wikipedia.org/wiki/Overengineering).

```
      new SuccessResponse(
        res,
        "Contests",
        { contests: contests.map(formatContest) },
      ).send();
```

Cách làm như trên làm tăng độ phức tạp của code, khiến những người lần đầu đọc code sẽ phải đọc toàn bộ phần implementation của `SuccessResponse` thì mới hiểu code trên làm gì.

Nếu mục đích chỉ là để đảm bảo format của response, nên viết:

```
res.send(formatResponse({
  msg: "Contest",
  data: { contests: contests.map(formatContest) }
}));
```

Code như vậy sẽ dễ hiểu hơn rất nhiều.

Hoặc tốt nhất, hãy giữ nguyên như cũ:

```
res.send({
  code: 0,
  msg: "Contest",
  data: {
    contests: contests.map(formatContest),
    server_time: commonUtils.dateToTimestamp(new Date()),
  }
});
```

Cách này dễ hiểu nhất.

Reply:
Tham khảo: [KISS principle](https://en.wikipedia.org/wiki/KISS_principle)